### PR TITLE
(On getgov-backup) Ticket #1305: Handle www. in availablity check

### LINKED ISSUE #1305: Handle www. prefix when input in /availability API
[PR#100 in getgov-home for additional context
](https://github.com/cisagov/getgov-home/pull/100)
Handle edge case of including [www](http://www/). in the search input. This is most likely best handled by updating the manage.get.gov's availability endpoint to handle the [www](http://www/). prefix when parsing arguments, similarly to how the API handles the .gov suffix.

Per Katherine:
 I envisioned that it would strip out the "www." when checking availability. So the confirmation message for "[www.example.gov](http://www.example.gov/)" would read: "[example.gov](http://example.gov/) is not available" Is that what you were thinking,

Example: 
 if [example.gov](http://example.gov/) was not available to begin with. I would think yes it strips www. then if [example.gov](http://example.gov/) is already taken it says “[example.gov](http://example.gov/) is not available”. If [example.gov](http://example.gov/) is actually available then entering [www.example.gov](http://www.example.gov/) would result in “[example.gov](http://example.gov/) is available”. Basically have it just ignore a www. at the start.

_Originally posted by @erinysong in https://github.com/cisagov/manage.get.gov/issues/476#issuecomment-1802870748_
            
[Slack thread](https://cisa-corp.slack.com/archives/C05BDEA3C11/p1705599697584059)

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #1305

## Changes
Adds a check for "www." and a unit test

## Context for reviewers
This ticket also[ requires this one](https://github.com/cisagov/get.gov/pull/206)
This PR changes how the check availablity check (and alternate domain check) works by stripping out the "www."

## Setup
**For designers**
Go to getgov-backup, and create a new domain request. Enter "www." into the domain name, such as "www.igortest"

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [x] Checked that the design translated visually
- [x] Checked behavior
- [x] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links
- [x] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [x] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/models/utility/domain_helper.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ def _validate_domain_string(domain, blank_ok):
             # If blank ok is true, just return the domain
             return domain
 
+        if domain.startswith("www."):
```
Comment (therealslimhsiehdy) [inline src/registrar/models/utility/domain_helper.py:60]: **Question (non-important, just random thought):** Now I'm wondering if we should have a case where maybe the user is copy pasta-ing and adds in the http(s) too... definitely interesting to see what our users will input

**Review by therealslimhsiehdy — APPROVED** [review]

**Comment by gabydisarli** [comment]
@zandercymatics I'll be doing the design review

File: src/registrar/models/utility/domain_helper.py (hunk 1 above)
Comment (gabydisarli) [reply src/registrar/models/utility/domain_helper.py:60]: I thought about this too, but hopefully they'll figure it out once they hit the error telling them not to use any weird characters.
Screenshot 2024-01-25 at 4 36 38 PM

**Review by gabydisarli — APPROVED** [review]
Looks great!

File: src/registrar/models/utility/domain_helper.py (hunk 1 above)
Comment (therealslimhsiehdy) [reply src/registrar/models/utility/domain_helper.py:60]: Great naming hahaha my cat is honoured 😆

**Pull request merged**